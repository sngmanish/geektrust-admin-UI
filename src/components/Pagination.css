.pagination-area {
    list-style: none;
    display: flex;
    margin: 0;
    padding: 0;
}

.page-no {
    height: 2rem;
    width: 2rem;
    padding: 0;
    margin: 0 0.5rem;
    border: 1px solid #00a8ff;
    vertical-align: center;
    border-radius: 50%;
    transition: 0.5s;
    background-color: hsl(200, 100%, 50%);
    color: #fff;
}

.page-no.active {
    cursor: pointer;
    border: 1px solid hsl(200, 100%, 50%);
    background-color: #fff;
    color: hsl(200, 100%, 50%);
}

.page-no.not-active {
    cursor: pointer;
    border: 1px solid #7f8fa6;
    color: #7f8fa6;
    background-color: #dcdde1
}

.page-no .pagination-content {
    text-align: center;
    font-size: 1rem;
    cursor: pointer;

}