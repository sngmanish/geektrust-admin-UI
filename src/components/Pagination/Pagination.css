/* .pagination-area {
  list-style: none;
  display: flex;
  margin: 0;
  padding: 0;
}

.page-no {
  height: 2rem;
  width: 2rem;
  padding: 0;
  margin: 0 0.25rem;
  border: 1px solid #00a8ff;
  vertical-align: center;
  border-radius: 50%;
  transition: 0.5s;
  background-color: hsl(200, 100%, 50%);
  color: #fff;
}

.page-no.active {
  cursor: pointer;
  border: 1px solid hsl(200, 100%, 50%);
  background-color: #fff;
  color: hsl(200, 100%, 50%);
}

.page-no.not-active {
  cursor: pointer;
  border: 1px solid #7f8fa6;
  color: #7f8fa6;
  background-color: #dcdde1;
}

.page-no .pagination-content {
  text-align: center;
  font-size: 1rem;
  cursor: pointer;
} */

/* .pagination {
  display: inline-block;
} */

.pagination {
  margin: 0px;
}
.pagination a {
  color: black;
  float: left;
  padding: 8px 16px;
  text-decoration: none;
  transition: background-color 0.3s;
}

.pagination a.active {
  background-color: #4caf50;
  color: white;
}

.pagination a:hover:not(.active) {
  background-color: #ddd;
}
