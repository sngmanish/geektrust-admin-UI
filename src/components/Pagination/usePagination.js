import { useEffect, useState } from 'react'

const usePagination = ({ initialPage = 1, initialPageSize = 10, onChange }) => {
  const [pagination, setPagination] = useState({
    page: initialPage,
    pageSize: initialPageSize
  })

  useEffect(() => {
    if (initialPage !== pagination.page) setPagination({ ...pagination, page: initialPage })
  }, [initialPage])

  useEffect(() => {
    if (initialPage !== pagination.pageSize)
      setPagination({ ...pagination, pageSize: initialPageSize })
  }, [initialPageSize])

  useEffect(() => {
    onChange && onChange(pagination.page, pagination.pageSize)
  }, [pagination.page, pagination.pageSize])

  const handlePageChange = page => setPagination({ ...pagination, page })
  const handlePageSizeChange = pageSize => setPagination({ ...pagination, page: 1, pageSize })

  return {
    page: pagination.page,
    pageSize: pagination.pageSize,
    handlePageChange,
    handlePageSizeChange
  }
}

export default usePagination
