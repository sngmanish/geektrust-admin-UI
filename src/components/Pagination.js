import React, { useEffect, useState } from "react";
import "./Pagination.css";
import config from "../config/index";

const Pagination = ({ page, setPage, totalNumberOfPages }) => {
    const pageLimit = config.MAX_PAGE;

    const [start, setstart] = useState(1);
    const [end, setEnd] = useState(Math.min(pageLimit, totalNumberOfPages));

    const getPaginationButtons = (start, end) => {
        return Array.from(
            { length: (end - start) / 1 + 1 },
            (_, i) => start + i * 1
        );
    };

    const [pageNum, setPageNum] = useState(getPaginationButtons(start, end, 1));

    const checkRange = () => {
        setPageNum(getPaginationButtons(start, totalNumberOfPages));
    };

    //TO FIRST PAGE
    const toFirstPage = () => {
        setPage(1);
    };

    //PREVIOUS PAGE
    const toPreviousPage = () => {
        setPage((page) => Math.max(page - 1, 1));
    };

    //1,2,3,4
    const customPage = (page) => {
        const pageNumber = Math.max(1, page);
        setPage(Math.min(pageNumber, totalNumberOfPages));
    };

    //NEXT PAGE
    const toNextPage = () => setPage(Math.min(page + 1, totalNumberOfPages));

    //LAST PAGE
    const toLastPage = () => setPage(totalNumberOfPages);

    useEffect(() => {
        checkRange();
    }, [totalNumberOfPages, page]);

    return (
        <>
            <ul className="pagination-area">
                <button
                    className={`page-no${page === 1 ? " not-active" : ""}`}
                    onClick={toFirstPage}
                >
                    <span className="pagination-content">
                        {`${String.fromCharCode(60)}${String.fromCharCode(60)}`}
                    </span>
                </button>
                <button
                    className={`page-no${page === 1 ? " not-active" : ""}`}
                    onClick={toPreviousPage}
                >
                    <span className="pagination-content">
                        {String.fromCharCode(60)}
                    </span>
                </button>

                {pageNum.map((item, index) => (
                    <button
                        key={index}
                        onClick={() => customPage(item)}
                        className={`page-no${page === item ? " active" : ""}`}
                    >
                        <span className="pagination-content">{item}</span>
                    </button>
                ))}
                <button
                    className={`page-no${
                        page === totalNumberOfPages ? " not-active" : ""
                    }`}
                    onClick={toNextPage}
                >
                    <span className="pagination-content">
                        {String.fromCharCode(62)}
                    </span>
                </button>
                <button
                    className={`page-no${
                        page === totalNumberOfPages ? " not-active" : ""
                    }`}
                    onClick={toLastPage}
                >
                    <span className="pagination-content">
                        {`${String.fromCharCode(62)}${String.fromCharCode(62)}`}
                    </span>
                </button>
            </ul>
        </>
    );
};

export default Pagination;
