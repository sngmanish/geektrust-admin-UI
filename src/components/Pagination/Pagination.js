import React, { useMemo } from 'react'
import './Pagination.css'
import config from '../../config/index'

// page onchange, count

const Pagination = ({ page, onChange, count }) => {
  const pageLimit = config.MAX_PAGE

  const pageNum = useMemo(
    () => Array.from({ length: Math.min(pageLimit, count) }, (_, i) => 1 + i * 1),
    [count, page]
  )

  const customPage = (e, page) => {
    const pageNumber = Math.max(1, page)
    onChange(e, Math.min(pageNumber, count))
  }

  return (
    <>
      <ul className="pagination">
        <a
          href="#"
          role="button"
          onClick={e => onChange(e, 1)}
          className={`${page === 1 ? ' disabled' : ''}`}
        >
          {'«'}
        </a>

        {pageNum.map((item, index) => (
          <a
            href="#"
            role="button"
            key={index}
            onClick={e => customPage(e, item)}
            className={`${page === item ? ' active' : ''}`}
          >
            {item}
          </a>
        ))}

        <a
          href="#"
          role="button"
          onClick={e => onChange(e, count)}
          className={`${page === count ? ' disabled' : ''}`}
        >
          {'»'}
        </a>
      </ul>
    </>
  )
}

export default Pagination
