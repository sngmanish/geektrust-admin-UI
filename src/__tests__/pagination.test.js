import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Pagination from '../components/Pagination/Pagination'

describe('Pagination', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Pagination page={1} onChange={() => {}} count={10} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('disables previous button when on first page', () => {
    const onChange = jest.fn()
    render(<Pagination page={1} onChange={onChange} count={10} />)
    const previousButton = screen.getByText('«')
    //     expect(previousButton).toHaveStyle(`
    //     pointer-events: none;
    //     color: gray;
    //     text-decoration: none;
    // `)
    fireEvent.click(previousButton)
    expect(onChange).toHaveBeenCalled()
  })

  it('disables next button when on last page', () => {
    const onChange = jest.fn()
    render(<Pagination page={10} onChange={onChange} count={10} />)
    const nextButton = screen.getByText('»')

    //     expect(nextButton).toHaveStyle(`
    //     pointer-events: none;
    //     color: gray;
    //     text-decoration: none;
    //   `)

    fireEvent.click(nextButton)
    expect(onChange).toHaveBeenCalled()
  })
})
