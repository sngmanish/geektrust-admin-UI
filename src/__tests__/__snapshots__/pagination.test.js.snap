// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<ul
  className="pagination"
>
  <a
    className=" disabled"
    href="#"
    onClick={[Function]}
    role="button"
  >
    «
  </a>
  <a
    className=" active"
    href="#"
    onClick={[Function]}
    role="button"
  >
    1
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
    role="button"
  >
    2
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
    role="button"
  >
    3
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
    role="button"
  >
    4
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
    role="button"
  >
    5
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
    role="button"
  >
    »
  </a>
</ul>
`;
